src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -940,7 +940,7 @@ else if (!getGuild().getSelfMember().hasPermission(permChan, Permission.MESSAGE_
             if (isFromType(ChannelType.PRIVATE))
                 throw new PermissionException("Cannot suppress embeds of others in a PrivateChannel.");
 
-            IPermissionContainer permChan = (IPermissionContainer) channel;
+            IPermissionContainer permChan = getPermissionContainer();
             if (!getGuild().getSelfMember().hasPermission(permChan, Permission.MESSAGE_MANAGE))
```

Comment:
Same statement as above. We don't need a concrete instance of `IPermissionContainer` here.